### PR TITLE
External/affiliate products are now editable when Products M3 flag is enabled

### PR DESCRIPTION
**Pull request by jaclync**
Editable external/affiliate product for #2200 

Note: Pending edit actions are SKU and external URL, as separate subtasks in #2200

## Changes

- Added `UIImage.linkImage` for the icon on the external URL row on the product form settings section
- Added `sku` and `externalURL` cases to `ProductFormEditAction` enum, and created the corresponding cell view models in `DefaultProductFormTableViewModel`. In `ProductFormActionsFactory`, updated the logic to return different actions for simple and affiliate products
- Added `editSKU` case to `ProductFormBottomSheetAction` enum, that is mapped from `ProductFormEditAction.sku`
- In `ProductDetailsFactory`, updated the logic to return a product form for an affiliate product if the Products M3 flag is enabled
- Updated unit tests

### Products M3 flag is on

- Go to the Products tab
- Tap on an external/affiliate Product without SKU --> should see at least price and external URL rows in the settings section
- Tap "Add more details" at the bottom --> should see at least "SKU" action in the bottom sheet (no-op for now)
- Dismiss the bottom sheet and try editing the affiliate product
- Tap "Update" --> the affiliate product should be updated remotely

### Products M3 flag is off

- Turn off the M3 feature flag in code
- Go to the Products tab
- Tap on an external/affiliate Product --> should see readonly product details

## Example screenshots

\ | product form actions | bottom sheet actions
-- | -- | --
Product has no SKU | Simulator Screen Shot - iPhone 11 - 2020-06-16 at 11 55 22 | Simulator Screen Shot - iPhone 11 - 2020-06-16 at 11 55 22
Product has a SKU | Simulator Screen Shot - iPhone 11 - 2020-06-16 at 11 55 42 | Simulator Screen Shot - iPhone 11 - 2020-06-16 at 11 55 50

Update release notes:
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by jaclync** [comment]
> I created a separate issue here but actually all the "empty field" should always be an action prompt not a descriptive label (so here, if there's no price yet then we should read "Add price" instead of "price")

@Garance91540 sounds good 👍 I'll address [this issue](https://github.com/woocommerce/woocommerce-ios/issues/2434) separately this sprint!

> I'm assuming the "Product type" row will be shown by default once we support product type change?

yes, the product type row & action will be implemented in https://github.com/woocommerce/woocommerce-ios/issues/2262 and https://github.com/woocommerce/woocommerce-ios/issues/2198 respectively.

> There's a small typo in the bottom sheet "Easily identify your products with unique codes"

great catch! I copied the string from the design without noticing the typo 😅 . here's an updated screenshot with the new copy (now it takes 2 lines on iPhone 11):

File: WooCommerce/WooCommerceTests/Mockups/MockProduct.swift [hunk 1]
```diff
@@ -18,6 +18,7 @@ final class MockProduct {
                  fullDescription: String? = "<p>This is the party room!</p>\n",
                  productShippingClass: ProductShippingClass? = nil,
                  backordersSetting: ProductBackordersSetting = .notAllowed,
+                 externalURL: String? = "http://somewhere.com",
```
Comment (pmusolino) [inline WooCommerce/WooCommerceTests/Mockups/MockProduct.swift:21]: nit: 
```suggestion
                 externalURL: String? = "https://example.com",
```

Comment (jaclync) [reply WooCommerce/WooCommerceTests/Mockups/MockProduct.swift]: updated in: https://github.com/woocommerce/woocommerce-ios/pull/2432/commits/d565e71f13a7da50b9ce91d470f9f894c93a70fe

File: WooCommerce/Classes/ViewRelated/Products/Edit Product/BottomSheetListSelector/ProductFormActionsFactory.swift [hunk 2]
```diff
@@ -65,14 +52,58 @@ struct ProductFormActionsFactory {
 
     /// Retruns an array of actions that are visible in the product form bottom sheet.
```
[the remark this answers is not in the document]
Comment (jaclync) [reply WooCommerce/Classes/ViewRelated/Products/Edit Product/BottomSheetListSelector/ProductFormActionsFactory.swift]: oops good catch! updated in: https://github.com/woocommerce/woocommerce-ios/pull/2432/commits/d565e71f13a7da50b9ce91d470f9f894c93a70fe

**Comment by jaclync** [comment]
thanks for the review Paolo, I addressed the changes you mentioned and I'm merging this PR now!

**Pull request merged**